apps/trading/components/swap/swap-form.tsx

```diff
@@ -0,0 +1,386 @@
+import { type FormEvent, useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import {
+  Intent,
+  TradingButton,
+  VegaIcon,
+  VegaIconNames,
+} from '@vegaprotocol/ui-toolkit';
+import { useVegaWallet } from '@vegaprotocol/wallet-react';
+import { useT } from '../../lib/use-t';
+import {
+  addDecimal,
+  removeDecimal,
+  roundUpToTickSize,
+  toBigNum,
+} from '@vegaprotocol/utils';
+import { OrderTimeInForce, OrderType, Side } from '@vegaprotocol/types';
+import { AssetInput, SwapButton, PriceImpactInput } from './swap-form-elements';
+import BigNumber from 'bignumber.js';
+import { Links } from '../../lib/links';
+import type { Account } from '@vegaprotocol/accounts';
+import type { AssetFieldsFragment } from '@vegaprotocol/assets';
+import {
+  getBaseAsset,
+  getQuoteAsset,
+  type MarketDataFieldsFragment,
+  type MarketFieldsFragment,
+} from '@vegaprotocol/markets';
+import { useVegaTransactionStore } from '@vegaprotocol/web3';
+import { getNotionalSize } from '@vegaprotocol/deal-ticket';
+import { usePrevious } from '@vegaprotocol/react-helpers';
+import { SpotData } from './spot-data';
+
+const getQuoteAssetBalance = (
+  asset?: AssetFieldsFragment,
+  accounts?: Account[] | null
+) => {
+  if (!asset || !accounts) return undefined;
+  const account = accounts.find((a) => a.asset.id === asset.id);
+  return account?.balance;
+};
+
+const derivePrice = (
+  marketData: MarketDataFieldsFragment | null,
+  side: Side,
+  toleranceFactor: number,
+  market: MarketFieldsFragment
+) => {
+  if (!marketData) return;
+  const isBid = side === Side.SIDE_SELL;
+  const price = isBid
+    ? new BigNumber(marketData.bestBidPrice).times(1 - toleranceFactor)
+    : new BigNumber(marketData.bestOfferPrice).times(1 + toleranceFactor);
+  return roundUpToTickSize(price, market.tickSize, isBid);
```

Comment:
Note, that `tickSize` is always associated with the QUOTE asset, and BASE is always associated with `positionDecimalPlaces` regardless of the side of the order.

Reply:
In other words, `tickSize` should be takes into consideration when changing the amount of QUOTE (regardless of "You pay" or "You receive")

And `positionDecimalPlaces` when changing the amount of BASE

Reply:
tickSize will be applied only on the market price